server_conn.go

```diff
@@ -333,6 +339,11 @@ func (c *serverConn) setUpgrading(name string, s transport.Server) {
 func (c *serverConn) upgraded() {
 	c.transportLocker.Lock()
 
+	if c.upgrading == nil {
```

Comment:
This is the actual fix - catches an out of band UPGRADE req from the client, and ignores it.

Also prints the IP address of the sender ... just in case you want to trace back to the source of a possible attack